In [15]:
import pandas as pd

## Load Raw Data 

The raw series and data CSV files are loaded using Latin-1 encoding because they could not be read correctly using Python’s default encoding.

In [16]:
series = pd.read_csv(
    "../data/food_prices_series_raw.csv",
    encoding="latin1"
)

series.head()

,Classification Name,Classification Code,Country Name,Country Code,Time,Time Code,Cost of an energy sufficient diet [CoCA],Cost of a nutrient adequate diet [CoNA],Cost of a healthy diet [CoHD],Cost of a healthy diet relative to the cost of sufficient energy from starchy staples [CoHD_CoCA],...,Affordability of an energy sufficient diet: ratio of cost to food expenditures [CoCA_fexp],Affordability of a nutrient adequate diet: ratio of cost to food expenditures [CoNA_fexp],Affordability of a healthy diet: ratio of cost to food expenditures [CoHD_fexp],Percent of the population who cannot afford sufficient calories [CoCA_headcount],Percent of the population who cannot afford nutrient adequacy [CoNA_headcount],Percent of the population who cannot afford a healthy diet [CoHD_headcount],Millions of people who cannot afford sufficient calories [CoCA_unafford_n],Millions of people who cannot afford nutrient adequacy [CoNA_unafford_n],Millions of people who cannot afford a healthy diet [CoHD_unafford_n],Population [Pop]
0,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2017.0,YR2017,0.604,1.88,3.476,5.75,...,0.111,0.346,0.64,0,3.2,21.8,0,0.4,2.5,11433438
1,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2018.0,YR2018,..,..,3.559,..,...,..,..,..,..,..,21.2,..,..,2.5,11565203
2,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2019.0,YR2019,..,..,3.628,..,...,..,..,..,..,..,20.8,..,..,2.4,11694721
3,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2020.0,YR2020,..,..,3.639,..,...,..,..,..,..,..,20.3,..,..,2.4,11818618
4,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2021.0,YR2021,..,..,..,..,...,..,..,..,..,..,..,..,..,..,..


In [17]:
data = pd.read_csv(
    "../data/food_prices_data_raw.csv",
    encoding="latin1"
)

data.head()

,Classification Name,Classification Code,Country Name,Country Code,Time,Time Code,Cost of an energy sufficient diet [CoCA],Cost of a nutrient adequate diet [CoNA],Cost of a healthy diet [CoHD],Cost of a healthy diet relative to the cost of sufficient energy from starchy staples [CoHD_CoCA],...,Affordability of an energy sufficient diet: ratio of cost to food expenditures [CoCA_fexp],Affordability of a nutrient adequate diet: ratio of cost to food expenditures [CoNA_fexp],Affordability of a healthy diet: ratio of cost to food expenditures [CoHD_fexp],Percent of the population who cannot afford sufficient calories [CoCA_headcount],Percent of the population who cannot afford nutrient adequacy [CoNA_headcount],Percent of the population who cannot afford a healthy diet [CoHD_headcount],Millions of people who cannot afford sufficient calories [CoCA_unafford_n],Millions of people who cannot afford nutrient adequacy [CoNA_unafford_n],Millions of people who cannot afford a healthy diet [CoHD_unafford_n],Population [Pop]
0,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2017.0,YR2017,0.604,1.88,3.476,5.75,...,0.111,0.346,0.64,0,3.2,21.8,0,0.4,2.5,11433438
1,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2018.0,YR2018,..,..,3.559,..,...,..,..,..,..,..,21.2,..,..,2.5,11565203
2,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2019.0,YR2019,..,..,3.628,..,...,..,..,..,..,..,20.8,..,..,2.4,11694721
3,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2020.0,YR2020,..,..,3.639,..,...,..,..,..,..,..,20.3,..,..,2.4,11818618
4,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2021.0,YR2021,..,..,..,..,...,..,..,..,..,..,..,..,..,..,..


## Convert Raw Data

The raw series and data CSV files are then converted into a working dataset

In [18]:
series.to_csv(
    "../data/food_prices_series_utf8.csv",
    index=False,
    encoding="utf-8"
)

In [19]:
series_utf8 = pd.read_csv("../data/food_prices_series_utf8.csv")
series_utf8.head()

,Classification Name,Classification Code,Country Name,Country Code,Time,Time Code,Cost of an energy sufficient diet [CoCA],Cost of a nutrient adequate diet [CoNA],Cost of a healthy diet [CoHD],Cost of a healthy diet relative to the cost of sufficient energy from starchy staples [CoHD_CoCA],...,Affordability of an energy sufficient diet: ratio of cost to food expenditures [CoCA_fexp],Affordability of a nutrient adequate diet: ratio of cost to food expenditures [CoNA_fexp],Affordability of a healthy diet: ratio of cost to food expenditures [CoHD_fexp],Percent of the population who cannot afford sufficient calories [CoCA_headcount],Percent of the population who cannot afford nutrient adequacy [CoNA_headcount],Percent of the population who cannot afford a healthy diet [CoHD_headcount],Millions of people who cannot afford sufficient calories [CoCA_unafford_n],Millions of people who cannot afford nutrient adequacy [CoNA_unafford_n],Millions of people who cannot afford a healthy diet [CoHD_unafford_n],Population [Pop]
0,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2017.0,YR2017,0.604,1.88,3.476,5.75,...,0.111,0.346,0.64,0,3.2,21.8,0,0.4,2.5,11433438
1,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2018.0,YR2018,..,..,3.559,..,...,..,..,..,..,..,21.2,..,..,2.5,11565203
2,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2019.0,YR2019,..,..,3.628,..,...,..,..,..,..,..,20.8,..,..,2.4,11694721
3,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2020.0,YR2020,..,..,3.639,..,...,..,..,..,..,..,20.3,..,..,2.4,11818618
4,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2021.0,YR2021,..,..,..,..,...,..,..,..,..,..,..,..,..,..,..


In [20]:
data.to_csv(
    "../data/food_prices_data_utf8.csv",
    index=False,
    encoding="utf-8"
)

In [21]:
data_utf8 = pd.read_csv("../data/food_prices_data_utf8.csv")
data_utf8.head()

,Classification Name,Classification Code,Country Name,Country Code,Time,Time Code,Cost of an energy sufficient diet [CoCA],Cost of a nutrient adequate diet [CoNA],Cost of a healthy diet [CoHD],Cost of a healthy diet relative to the cost of sufficient energy from starchy staples [CoHD_CoCA],...,Affordability of an energy sufficient diet: ratio of cost to food expenditures [CoCA_fexp],Affordability of a nutrient adequate diet: ratio of cost to food expenditures [CoNA_fexp],Affordability of a healthy diet: ratio of cost to food expenditures [CoHD_fexp],Percent of the population who cannot afford sufficient calories [CoCA_headcount],Percent of the population who cannot afford nutrient adequacy [CoNA_headcount],Percent of the population who cannot afford a healthy diet [CoHD_headcount],Millions of people who cannot afford sufficient calories [CoCA_unafford_n],Millions of people who cannot afford nutrient adequacy [CoNA_unafford_n],Millions of people who cannot afford a healthy diet [CoHD_unafford_n],Population [Pop]
0,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2017.0,YR2017,0.604,1.88,3.476,5.75,...,0.111,0.346,0.64,0,3.2,21.8,0,0.4,2.5,11433438
1,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2018.0,YR2018,..,..,3.559,..,...,..,..,..,..,..,21.2,..,..,2.5,11565203
2,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2019.0,YR2019,..,..,3.628,..,...,..,..,..,..,..,20.8,..,..,2.4,11694721
3,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2020.0,YR2020,..,..,3.639,..,...,..,..,..,..,..,20.3,..,..,2.4,11818618
4,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2021.0,YR2021,..,..,..,..,...,..,..,..,..,..,..,..,..,..,..


## Inspect the Working DataSet Files

Before cleaning the data, I inspect the first several rows, dimensions, and column names of both raw files. This helps me understand how the data are organized, identify the variables relevant to the research question, and check for issues such as unexpected column names or missing values.

In [22]:
print(data.shape)
display(data.head())

print(series.shape)
display(series.head())

(3725, 40)


,Classification Name,Classification Code,Country Name,Country Code,Time,Time Code,Cost of an energy sufficient diet [CoCA],Cost of a nutrient adequate diet [CoNA],Cost of a healthy diet [CoHD],Cost of a healthy diet relative to the cost of sufficient energy from starchy staples [CoHD_CoCA],...,Affordability of an energy sufficient diet: ratio of cost to food expenditures [CoCA_fexp],Affordability of a nutrient adequate diet: ratio of cost to food expenditures [CoNA_fexp],Affordability of a healthy diet: ratio of cost to food expenditures [CoHD_fexp],Percent of the population who cannot afford sufficient calories [CoCA_headcount],Percent of the population who cannot afford nutrient adequacy [CoNA_headcount],Percent of the population who cannot afford a healthy diet [CoHD_headcount],Millions of people who cannot afford sufficient calories [CoCA_unafford_n],Millions of people who cannot afford nutrient adequacy [CoNA_unafford_n],Millions of people who cannot afford a healthy diet [CoHD_unafford_n],Population [Pop]
0,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2017.0,YR2017,0.604,1.88,3.476,5.75,...,0.111,0.346,0.64,0,3.2,21.8,0,0.4,2.5,11433438
1,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2018.0,YR2018,..,..,3.559,..,...,..,..,..,..,..,21.2,..,..,2.5,11565203
2,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2019.0,YR2019,..,..,3.628,..,...,..,..,..,..,..,20.8,..,..,2.4,11694721
3,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2020.0,YR2020,..,..,3.639,..,...,..,..,..,..,..,20.3,..,..,2.4,11818618
4,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2021.0,YR2021,..,..,..,..,...,..,..,..,..,..,..,..,..,..,..


(3760, 40)


,Classification Name,Classification Code,Country Name,Country Code,Time,Time Code,Cost of an energy sufficient diet [CoCA],Cost of a nutrient adequate diet [CoNA],Cost of a healthy diet [CoHD],Cost of a healthy diet relative to the cost of sufficient energy from starchy staples [CoHD_CoCA],...,Affordability of an energy sufficient diet: ratio of cost to food expenditures [CoCA_fexp],Affordability of a nutrient adequate diet: ratio of cost to food expenditures [CoNA_fexp],Affordability of a healthy diet: ratio of cost to food expenditures [CoHD_fexp],Percent of the population who cannot afford sufficient calories [CoCA_headcount],Percent of the population who cannot afford nutrient adequacy [CoNA_headcount],Percent of the population who cannot afford a healthy diet [CoHD_headcount],Millions of people who cannot afford sufficient calories [CoCA_unafford_n],Millions of people who cannot afford nutrient adequacy [CoNA_unafford_n],Millions of people who cannot afford a healthy diet [CoHD_unafford_n],Population [Pop]
0,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2017.0,YR2017,0.604,1.88,3.476,5.75,...,0.111,0.346,0.64,0,3.2,21.8,0,0.4,2.5,11433438
1,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2018.0,YR2018,..,..,3.559,..,...,..,..,..,..,..,21.2,..,..,2.5,11565203
2,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2019.0,YR2019,..,..,3.628,..,...,..,..,..,..,..,20.8,..,..,2.4,11694721
3,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2020.0,YR2020,..,..,3.639,..,...,..,..,..,..,..,20.3,..,..,2.4,11818618
4,Food Prices for Nutrition 1.0,FPN 1.0,Tunisia,TUN,2021.0,YR2021,..,..,..,..,...,..,..,..,..,..,..,..,..,..,..


In [23]:
print(data.columns.tolist())
print(series.columns.tolist())

['Classification Name', 'Classification Code', 'Country Name', 'Country Code', 'Time', 'Time Code', 'Cost of an energy sufficient diet [CoCA]', 'Cost of a nutrient adequate diet [CoNA]', 'Cost of a healthy diet [CoHD]', 'Cost of a healthy diet relative to the cost of sufficient energy from starchy staples [CoHD_CoCA]', 'Cost of fruits [CoHD_f]', 'Cost of vegetables [CoHD_v]', 'Cost of starchy staples [CoHD_ss]', 'Cost of animal-source foods [CoHD_asf]', 'Cost of legumes, nuts and seeds [CoHD_lns]', 'Cost of oils and fats [CoHD_of]', 'Cost share for fruits in a least-cost healthy diet [CoHD_f_prop]', 'Cost share for vegetables in a least-cost healthy diet [CoHD_v_prop]', 'Cost share for starchy staples in a least-cost healthy diet [CoHD_ss_prop]', 'Cost share for animal-sourced foods in a least-cost healthy diet [CoHD_asf_prop]', 'Cost share for legumes, nuts and seeds in a least-cost healthy diet [CoHD_lns_prop]', 'Cost share for oils and fats in a least-cost healthy diet [CoHD_of_prop

## Shorten Column Names

The original dataset contains long column names, especially for the diet-cost variables. I rename the columns used in the analysis with shorter, clearer names such as `country`, `year`, `cost_energy`, `cost_nutrient`, and `cost_healthy`. These changes are made only in the working dataset, so the original raw data remain unchanged.

In [24]:
data_clean = data.rename(columns={
    "Classification Name": "classification",
    "Classification Code": "classification_code",
    "Country Name": "country",
    "Country Code": "country_code",
    "Time": "year",
    "Time Code": "year_code",

    "Cost of an energy sufficient diet [CoCA]": "cost_energy",
    "Cost of a nutrient adequate diet [CoNA]": "cost_nutrient",
    "Cost of a healthy diet [CoHD]": "cost_healthy",
})

In [25]:
important_columns = [
    "country",
    "country_code",
    "classification",
    "year",
    "cost_energy",
    "cost_nutrient",
    "cost_healthy",
]

In [26]:
diet_data = data_clean[important_columns].copy()

diet_data.head()

,country,country_code,classification,year,cost_energy,cost_nutrient,cost_healthy
0,Tunisia,TUN,Food Prices for Nutrition 1.0,2017.0,0.604,1.88,3.476
1,Tunisia,TUN,Food Prices for Nutrition 1.0,2018.0,..,..,3.559
2,Tunisia,TUN,Food Prices for Nutrition 1.0,2019.0,..,..,3.628
3,Tunisia,TUN,Food Prices for Nutrition 1.0,2020.0,..,..,3.639
4,Tunisia,TUN,Food Prices for Nutrition 1.0,2021.0,..,..,..


## Prepare the Diet-Cost Data for Analysis

I convert the three diet-cost columns to numeric values, treating nonnumeric entries such as `..` as missing data. Because 2017 is the only year with complete coverage across countries, I filter the dataset to that year and filter out missing costs. I updated the title of the diet categories. I then reshape the data from wide format to long format so that each row represents one country and one diet type. 

In [27]:
cost_columns = [
    "cost_energy",
    "cost_nutrient",
    "cost_healthy"
]

diet_data[cost_columns] = diet_data[cost_columns].apply(
    pd.to_numeric,
    errors="coerce"
)

In [28]:
diet_data["year"].unique()

array([2017., 2018., 2019., 2020., 2021.,   nan])

In [29]:
diet_2017 = diet_data[diet_data["year"] == 2017].copy()

In [30]:
diet_long = diet_2017.melt(
    id_vars=["classification", "country", "country_code", "year"],
    value_vars=["cost_energy", "cost_nutrient", "cost_healthy"],
    var_name="diet_type",
    value_name="cost"
)

In [31]:
diet_long["diet_type"] = diet_long["diet_type"].replace({
    "cost_energy": "Energy sufficient",
    "cost_nutrient": "Nutrient adequate",
    "cost_healthy": "Healthy"
})

In [32]:
diet_long = diet_long.dropna(subset=["cost"])

In [33]:
diet_long.head()

,classification,country,country_code,year,diet_type,cost
0,Food Prices for Nutrition 1.0,Tunisia,TUN,2017.0,Energy sufficient,0.604
1,Food Prices for Nutrition 1.0,Algeria,DZA,2017.0,Energy sufficient,0.773
2,Food Prices for Nutrition 1.0,Mauritania,MRT,2017.0,Energy sufficient,0.877
3,Food Prices for Nutrition 1.0,Cameroon,CMR,2017.0,Energy sufficient,0.894
4,Food Prices for Nutrition 1.0,Central African Republic,CAF,2017.0,Energy sufficient,1.021


## Check Cleaned Data 

There are duplicates in my dataset, as well as countries with incomplete diet-cost data. I check whether each country–year–diet-type combination appears more than once and then remove duplicate observations.

In [34]:
diet_long["country"].sort_values().unique()

array(['Albania', 'Algeria', 'Angola', 'Anguilla', 'Antigua and Barbuda',
       'Argentina', 'Armenia', 'Aruba', 'Australia', 'Austria',
       'Azerbaijan', 'Bahamas, The', 'Bahrain', 'Bangladesh', 'Barbados',
       'Belarus', 'Belgium', 'Belize', 'Benin', 'Bermuda', 'Bhutan',
       'Bolivia', 'Bonaire', 'Bosnia and Herzegovina', 'Botswana',
       'Brazil', 'British Virgin Islands', 'Brunei Darussalam',
       'Bulgaria', 'Burkina Faso', 'Burundi', 'Cabo Verde', 'Cambodia',
       'Cameroon', 'Canada', 'Cayman Islands', 'Central African Republic',
       'Chad', 'Chile', 'China', 'Colombia', 'Comoros',
       'Congo, Dem. Rep.', 'Congo, Rep.', 'Costa Rica', 'Croatia',
       'Curacao', 'Cyprus', 'Czech Republic', "Côte d'Ivoire", 'Denmark',
       'Djibouti', 'Dominica', 'Dominican Republic',
       'East Asia & Pacific', 'Ecuador', 'Egypt, Arab Rep.',
       'El Salvador', 'Equatorial Guinea', 'Estonia', 'Eswatini',
       'Ethiopia', 'Europe & Central Asia', 'Fiji', 'Finland', '

In [65]:
print("Number of rows:", diet_long.shape[0])
print("Number of countries:", diet_long["country"].nunique())

print("\nRows per diet type:")
print(diet_long["diet_type"].value_counts())

print("\nMissing values:")
print(diet_long.isna().sum())

Number of rows: 553
Number of countries: 186

Rows per diet type:
diet_type
Energy sufficient    186
Nutrient adequate    186
Healthy              181
Name: count, dtype: int64

Missing values:
classification    0
country           0
country_code      0
year              0
diet_type         0
cost              0
dtype: int64


In [66]:
print("Years:")
print(diet_long["year"].value_counts().sort_index())

print("\nClassifications:")
print(diet_long["classification"].value_counts())

Years:
year
2017.0    553
Name: count, dtype: int64

Classifications:
classification
Food Prices for Nutrition 2.1    553
Name: count, dtype: int64


In [67]:
diet_long[
    diet_long["country"] == "Tunisia"
].sort_values(["year", "classification", "diet_type"])

,classification,country,country_code,year,diet_type,cost
558,Food Prices for Nutrition 2.1,Tunisia,TUN,2017.0,Energy sufficient,0.604
2046,Food Prices for Nutrition 2.1,Tunisia,TUN,2017.0,Healthy,3.476
1302,Food Prices for Nutrition 2.1,Tunisia,TUN,2017.0,Nutrient adequate,1.880


In [38]:
diet_long.groupby(
    ["country", "year", "diet_type"]
).size().value_counts()

4    553
Name: count, dtype: int64

In [39]:
diet_long = diet_long[
    diet_long["classification"] == "Food Prices for Nutrition 2.1"
].copy()

In [40]:
print("Number of rows:", diet_long.shape[0])
print("Number of countries:", diet_long["country"].nunique())

print("\nRows per diet type:")
print(diet_long["diet_type"].value_counts())

print("\nClassifications:")
print(diet_long["classification"].value_counts())

Number of rows: 553
Number of countries: 186

Rows per diet type:
diet_type
Energy sufficient    186
Nutrient adequate    186
Healthy              181
Name: count, dtype: int64

Classifications:
classification
Food Prices for Nutrition 2.1    553
Name: count, dtype: int64


In [41]:
complete_countries = (
    diet_long
    .groupby("country")["diet_type"]
    .nunique()
)

complete_countries = complete_countries[
    complete_countries == 3
].index

diet_long_complete = diet_long[
    diet_long["country"].isin(complete_countries)
].copy()

In [42]:
print("Number of rows:", diet_long_complete.shape[0])
print("Number of countries:", diet_long_complete["country"].nunique())

print("\nRows per diet type:")
print(diet_long_complete["diet_type"].value_counts())

Number of rows: 543
Number of countries: 181

Rows per diet type:
diet_type
Energy sufficient    181
Nutrient adequate    181
Healthy              181
Name: count, dtype: int64


## Save the Cleaned Dataset

I save the final cleaned dataset for use in the analysis and visualizations. It contains 2017 cost data for 181 countries with complete observations for all three diet types.

In [43]:
diet_long_complete.to_csv(
    "../data/diet_costs_2017_clean.csv",
    index=False
)

## Create Affordability Dataset

In this section, I create a separate cleaned dataset for affordability. I keep the country identifiers, year, dataset version, affordability ratios, and percentage of the population unable to afford each diet type. I then filter the data to 2017 and Food Prices for Nutrition version 2.1, convert the affordability variables to numeric values, and remove countries with missing affordability data. The final affordability dataset contains countries with complete affordability information for energy-sufficient, nutrient-adequate, and healthy diets.

In [44]:
import pandas as pd

In [69]:
raw_data = data_utf8.copy()

In [70]:
raw_data.columns.tolist()

['Classification Name',
 'Classification Code',
 'Country Name',
 'Country Code',
 'Time',
 'Time Code',
 'Cost of an energy sufficient diet [CoCA]',
 'Cost of a nutrient adequate diet [CoNA]',
 'Cost of a healthy diet [CoHD]',
 'Cost of a healthy diet relative to the cost of sufficient energy from starchy staples [CoHD_CoCA]',
 'Cost of fruits [CoHD_f]',
 'Cost of vegetables [CoHD_v]',
 'Cost of starchy staples [CoHD_ss]',
 'Cost of animal-source foods [CoHD_asf]',
 'Cost of legumes, nuts and seeds [CoHD_lns]',
 'Cost of oils and fats [CoHD_of]',
 'Cost share for fruits in a least-cost healthy diet [CoHD_f_prop]',
 'Cost share for vegetables in a least-cost healthy diet [CoHD_v_prop]',
 'Cost share for starchy staples in a least-cost healthy diet [CoHD_ss_prop]',
 'Cost share for animal-sourced foods in a least-cost healthy diet [CoHD_asf_prop]',
 'Cost share for legumes, nuts and seeds in a least-cost healthy diet [CoHD_lns_prop]',
 'Cost share for oils and fats in a least-cost healt

In [75]:
afford_data = raw_data[
    [
        "Classification Name",
        "Country Name",
        "Country Code",
        "Time",
        "Affordability of an energy sufficient diet: ratio of cost to food expenditures [CoCA_fexp]",
        "Affordability of a nutrient adequate diet: ratio of cost to food expenditures [CoNA_fexp]",
        "Affordability of a healthy diet: ratio of cost to food expenditures [CoHD_fexp]",
        "Percent of the population who cannot afford sufficient calories [CoCA_headcount]",
        "Percent of the population who cannot afford nutrient adequacy [CoNA_headcount]",
        "Percent of the population who cannot afford a healthy diet [CoHD_headcount]"
    ]
].copy()

In [79]:
afford_data = affordability_data.rename(columns={
    "Classification Name": "classification",
    "Country Name": "country",
    "Country Code": "country_code",
    "Time": "year",

    "Affordability of an energy sufficient diet: ratio of cost to food expenditures [CoCA_fexp]":
        "afford_energy",

    "Affordability of a nutrient adequate diet: ratio of cost to food expenditures [CoNA_fexp]":
        "afford_nutrient",

    "Affordability of a healthy diet: ratio of cost to food expenditures [CoHD_fexp]":
        "afford_healthy",

    "Percent of the population who cannot afford sufficient calories [CoCA_headcount]":
        "pct_cannot_afford_energy",

    "Percent of the population who cannot afford nutrient adequacy [CoNA_headcount]":
        "pct_cannot_afford_nutrient",

    "Percent of the population who cannot afford a healthy diet [CoHD_headcount]":
        "pct_cannot_afford_healthy"
})

In [80]:
afford_data.columns.tolist()

['classification',
 'country',
 'country_code',
 'year',
 'afford_energy',
 'afford_nutrient',
 'afford_healthy',
 'pct_cannot_afford_energy',
 'pct_cannot_afford_nutrient',
 'pct_cannot_afford_healthy']

In [83]:
afford_keep = [
    "classification",
    "country",
    "country_code",
    "year",
    "afford_energy",
    "afford_nutrient",
    "afford_healthy",
    "pct_cannot_afford_energy",
    "pct_cannot_afford_nutrient",
    "pct_cannot_afford_healthy"
]

afford_data = afford_data[afford_keep].copy()

In [93]:
afford_data = afford_data[
    (afford_data["year"] == 2017) &
    (afford_data["classification"] == "Food Prices for Nutrition 2.1")
].copy()

In [94]:
afford_data.head()

,classification,country,country_code,year,afford_energy,afford_nutrient,afford_healthy,pct_cannot_afford_energy,pct_cannot_afford_nutrient,pct_cannot_afford_healthy
2790,Food Prices for Nutrition 2.1,Tunisia,TUN,2017.0,0.111,0.346,0.640,0.0,1.9,15.5
2795,Food Prices for Nutrition 2.1,Algeria,DZA,2017.0,0.124,0.374,0.605,0.1,6.2,32.4
2800,Food Prices for Nutrition 2.1,Mauritania,MRT,2017.0,0.263,0.845,1.036,2.7,48.3,61.7
2805,Food Prices for Nutrition 2.1,Cameroon,CMR,2017.0,0.385,0.890,1.128,15.7,48.6,59.2
2810,Food Prices for Nutrition 2.1,Central African Republic,CAF,2017.0,0.823,1.376,2.761,64.1,83.3,95.7


Make sure the values are numeric

In [95]:
afford_columns = [
    "afford_energy",
    "afford_nutrient",
    "afford_healthy",
    "pct_cannot_afford_energy",
    "pct_cannot_afford_nutrient",
    "pct_cannot_afford_healthy"
]

afford_data[afford_columns] = afford_data[afford_columns].apply(
    pd.to_numeric,
    errors="coerce"
)

In [96]:
afford_data.isna().sum()

classification                 0
country                        0
country_code                   0
year                           0
afford_energy                  0
afford_nutrient                0
afford_healthy                 5
pct_cannot_afford_energy      31
pct_cannot_afford_nutrient    31
pct_cannot_afford_healthy     31
dtype: int64

In [102]:
afford_data_clean = afford_data.dropna(
    subset=afford_columns
).copy()

In [103]:
afford_data_clean.isna().sum()

classification                0
country                       0
country_code                  0
year                          0
afford_energy                 0
afford_nutrient               0
afford_healthy                0
pct_cannot_afford_energy      0
pct_cannot_afford_nutrient    0
pct_cannot_afford_healthy     0
dtype: int64

In [104]:
print("Rows before:", afford_data.shape[0])
print("Rows after:", afford_data_clean.shape[0])
print("Countries after:", afford_data_clean["country"].nunique())

Rows before: 186
Rows after: 155
Countries after: 155


In [106]:
afford_data_clean.head()

,classification,country,country_code,year,afford_energy,afford_nutrient,afford_healthy,pct_cannot_afford_energy,pct_cannot_afford_nutrient,pct_cannot_afford_healthy
2790,Food Prices for Nutrition 2.1,Tunisia,TUN,2017.0,0.111,0.346,0.640,0.0,1.9,15.5
2795,Food Prices for Nutrition 2.1,Algeria,DZA,2017.0,0.124,0.374,0.605,0.1,6.2,32.4
2800,Food Prices for Nutrition 2.1,Mauritania,MRT,2017.0,0.263,0.845,1.036,2.7,48.3,61.7
2805,Food Prices for Nutrition 2.1,Cameroon,CMR,2017.0,0.385,0.890,1.128,15.7,48.6,59.2
2810,Food Prices for Nutrition 2.1,Central African Republic,CAF,2017.0,0.823,1.376,2.761,64.1,83.3,95.7


In [108]:
afford_data_clean.duplicated(
    subset=["country", "country_code"]
).sum()

0

## Save the Cleaned Dataset

In [110]:
afford_data_clean.to_csv(
    "../data/affordability_2017_clean.csv",
    index=False
)